src/thor/trace_route_action.cc

```diff
@@ -386,58 +388,55 @@ thor_worker_t::map_match(Api& request, uint32_t best_paths) {
     if (options.action() == Options::trace_attributes) {
       // we make a new route to hold the result of each iteraton of top k
       auto& route = *request.mutable_trip()->mutable_routes()->Add();
-      path_map_match(match_results, path_edges, *route.mutable_legs()->Add(), route_discontinuities);
+      path_map_match(match_results, m_temp_path_edges, *route.mutable_legs()->Add(),
+                     m_temp_route_discontinuities);
     } // trace_route can return multiple trip paths and cannot have discontinuities
     else {
+
       // we make a new route to hold the result of each iteraton of top k
       auto& route = *request.mutable_trip()->mutable_routes()->Add();
       auto origin = match_results.begin();
       auto destination = match_results.begin();
+      int last_leg_index = 0;
 
-      size_t last_index = 0;
-      // for each edge in the path
-      for (size_t i = 0; i < path_edges.size(); ++i) {
-        const auto& path_edge = path_edges[i];
-        // while we still have trace points that are on this current edge of the path
-        while (destination != match_results.end() && path_edge.edgeid == destination->edgeid) {
-          // if it's not the first trace point, and it's a break, make a trip path leg
-          if (origin != destination && options.shape(destination - match_results.begin()).type() ==
```

Comment:
for people who dont know what this code is trying to do, i would argue that each comment here is useful at summarizing what is taking place. by removing these comments it makes it harder for experts to understand what is going on in the code. if the comments are not accurate, please update them but otherwise please do not delete them unless they are describing something trivial (the first comment is pretty trivial)